patches/minecraft/net/minecraft/block/BlockRedstoneComparator.java.patch

```diff
@@ -7,7 +7,7 @@
 +    @Override
 +    public void onNeighborChange(IBlockAccess world, BlockPos pos, BlockPos neighbor)
 +    {
-+        if (pos.func_177956_o() == neighbor.func_177956_o() && world instanceof World)
++        if (pos.func_177956_o() == neighbor.func_177956_o() && world instanceof net.minecraft.world.WorldServer)
```

Comment:
So the proper solution would be to have out patch to removeTileEntity have a isRemote check above the call to updateCompator not to hack the remote check in here.